src/core/models/highlightedvalue.js

```diff
@@ -135,6 +144,56 @@ export default class HighlightedValue {
     return highlightedValue;
   }
 
+  /**
+   * introduces highlighting to input data according to highlighting specifiers and returns a
+   * list of elements/strings
+   *
+   * @param {Object} val input object to apply highlighting to
+   * @param {Object} highlightedSubstrings highlighting specifiers to apply to input object
+   * @param {Function} transformFunction function to apply to strings in between highlighting markup
+   * @returns {Array<Element|string>} a list of elements/strings representing the val with
+   * highlighting applied. Example returned value :
+   *   [
+   *     'Save time & bank on your terms at over 1,800',
+   *     <strong>ATM</strong>,
+   *   ]
+   */
+  buildHighlightedElements (
```

Comment:
this function looks similar to buildHighlightedValue above. what's the difference between the two? just the return type? is it possible to reuse some of the logic between them?

Reply:
Yeah. I'd prefer to keep them separate since the return type and how to construct it is quite different. Keeping them separate can be clearer.